### DISCUSSION TITLE
Graph Version is always 3.1 for the right hemisphere and 3.3 for the left hemisphere

### DISCUSSION THREAD

**Discussion by wsnyder4** [brainvisa/brainvisa.github.io#116] [Q&A] [answered]
Dear BrainVISA team,

I have compiled BrainVISA using the latest bugfix version that has allowed me to take advantage of the bug fix that makes sure all CNN labeled graphs have version 3.3 (https://github.com/brainvisa/morpho-deepsulci/issues/7). 
However, after the fix I still get  graphs are version 3.1 for the right hemisphere, although the left graphs that were previously version 5.0 are now 3.3.

How does graph version affect the results? Would it be okay to just have my CNN labeled graphs be version 3.1 and keep it consistent as 3.1?

Thank you for your help!
All the best,
Will

**Comment by denisri** [comment, marked as answer]
Hi Will,
So you get graphs version 3.3 on the left side and 3.1 on the right side ? How strange, I can't reproduce this behaviour.
3.1 and 3.3 only differ in the thickness of the junction "lines" (voxels lists), thus it does not impact much. Most morphometric measurements do not rely on this, except a few relations measurements (and even there, most measurements such as direction should not change significantly). But this asymetrical behaviour is really strange...

**Reply by wsnyder4** [reply]
Hi Denis,

Thanks for your reply and just wanted to follow up again. I had brushed this off as I figured I had maybe done something wrong and since it seemed to work well to just keep both graph versions set to 3.1 for my purposes. However, in working with Heloise, we are now seeing that in her BrainVISA install (also using casa-dev version 5.1 with a recent compiling from github) that the asymmetry is now showing up again -- specifically, when a graph is loaded into anatomist (or read with pyaims), the left hemisphere is always v 3.3 and the right is v 3.1.  

Before the bug fix that forced the graph version after CNN recognition to be 3.3 (after originally it was showing 5.0), I had the same asymmetry but with the left hemisphere being 5.0 and the right being v 3.1. So potentially somewhere the right hemisphere is just being forced again to be 3.1 still (or isn't changed from 3.1?). In trying to replicate the mean / max depth measure from brainvisa, I noticed that this only replicated the values from the sulcal morphometry csv when the graph output by CNN recognition was 3.1 (hence why I force graph version to be 3.1 in my install currently). 

Are sulcal skeleton images or the sulcal graphs edited by the CNN recognition approach? Because possibly the csv files report values based on skeleton images or graphs before the recognition approach changes the graph or skeleton image? I am not sure, but am curious whether this would effect classic brainvisa morphometry reporting for each hemisphere or just seems to pop up in my use case for processing depth following CNN recogniton.

Sorry for the long message but just wanted to put out all my thoughts in case any are useful. Thank you!

All the best,
Will

**Reply by denisri** [reply]
I don't understand the systematic asymmetry, but I may have a clue for a variable behaviour. I have to check that anyway because I don't remember exactly everything and I have not coded that part myself, but: in the CNN recognition, some graph nodes may be split into several ones when the whole node should not be labelled the same way. It doesn't happen very often (generally a few times in a hemisphere) and sometimes it doesn't occur at all in a graph. When it occurs, then the graph structure is changed, and the structure needs to be adapted and things recalculated. Normally this is the step which switches graphs to version 3.3. But if no node is split, then this step is possibly not performed at all, and the initial graph version (which should be 3.1 I guess) will be maintained. But I'm surprised that it systematically occurs on one side and never on the other, that is wrong...
The only difference between 3.1 and 3.3 is that junction voxels lists (in relations between nodes) are thinner in 3.3 than they used to be in 3.1. The rest should be the same.
Denis
